src/Core/FeatureFlag/FeatureFlagManager.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag;
+
+use PrestaShop\PrestaShop\Core\FeatureFlag\Layer\DbLayer;
+use PrestaShop\PrestaShop\Core\FeatureFlag\Layer\DotEnvLayer;
+use PrestaShop\PrestaShop\Core\FeatureFlag\Layer\EnvLayer;
+use PrestaShopBundle\Entity\Repository\FeatureFlagRepository;
+use Psr\Container\ContainerInterface;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+class FeatureFlagManager implements ServiceSubscriberInterface
```

Comment:
```suggestion
class FeatureFlagManager implements ServiceSubscriberInterface, TypeLayerInterface
```
Shouldn't this one implement the interface as well? This way you could define a service based on the interface, the default alias would this one and everywhere you can simply get the `TypeLayerInterface` service from the container